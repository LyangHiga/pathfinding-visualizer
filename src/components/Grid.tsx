import React from "react";

import Node from "./Node";

import { GridProps } from "./types/GridTypes";

function Grid(props: GridProps) {
  const {
    grid,
    isWeighted,
    handleMouseDown,
    handleMouseEnter,
    handleMouseUp,
  } = props;

  return (
    <div style={{ textAlign: "center" }}>
      {grid.grid.map((row, rowIdx) => {
        return (
          <div key={rowIdx}>
            {row.map((node) => {
              const { row, col, val, weight } = node;
              return (
                <Node
                  key={val}
                  val={val}
                  col={col}
                  row={row}
                  onMouseDown={handleMouseDown}
                  onMouseEnter={handleMouseEnter}
                  onMouseUp={handleMouseUp}
                  weight={weight}
                  isWeighted={isWeighted}
                ></Node>
              );
            })}
          </div>
        );
      })}
    </div>
  );
}

export default Grid;
