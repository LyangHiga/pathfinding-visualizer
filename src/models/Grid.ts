import Node from "./Node";
import { createAdjList, getVal } from "../helpers/gridHelper";

export default class Grid {
  public grid: Node[][];
  public start: number;
  public target: number;
  public readonly nRows: number;
  public readonly nCols: number;
  // min and max weight allowed
  public readonly min: number;
  public readonly max: number;
  constructor(
    start: number,
    target: number,
    nRows: number,
    nCols: number,
    // min and max weight
    max = 1,
    min = 1
  ) {
    this.grid = [];
    this.start = start;
    this.target = target;
    this.nCols = nCols;
    this.nRows = nRows;
    this.min = min;
    this.max = max;
    for (let row = 0; row < nRows; row++) {
      const currentRow: Node[] = [];
      for (let col = 0; col < nCols; col++) {
        // get random weight for all edges of this node
        const weight = Math.floor(Math.random() * (max - min) + min);
        const val = getVal(row, nCols, col);
        const AdjList = createAdjList(val, col, row, nRows, nCols);
        const newNode = new Node(col, row, val, start, target, AdjList, weight);
        currentRow.push(newNode);
      }
      this.grid.push(currentRow);
    }
  }
}
