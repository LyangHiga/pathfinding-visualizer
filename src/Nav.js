// TODO: Refactoring Drawer and btns
import React, { Fragment, useState } from "react";
import "rc-slider/assets/index.css";
import bfs from "./algorithms/bfs";
import dfs from "./algorithms/dfs";
import a from "./algorithms/a";
import bellmanFord from "./algorithms/bellmanFord";
import { clearAnimation, clearPathAnimation } from "./helpers/animations";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";

import Slider from "rc-slider";
import AppBar from "@material-ui/core/AppBar";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { useTheme } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import SwipeableDrawer from "@material-ui/core/SwipeableDrawer";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";

import styles from "./NavStyles";
// import { getNewMazedGrid } from "./helpers/gridPropertiesHelper";
import { valToIndx, getNewMazedGrid } from "./helpers/gridHelper";

import Grid from "./models/Grid";

const MIN = -5;

function Nav(props) {
  const {
    grid,
    disable,
    start,
    end,
    setGrid,
    setDisable,
    nRows,
    nCols,
    wRange,
    isWeighted,
    setIsWeighted,
    toggleIsweighted,
    isNegative,
    toggleIsNegative,
    handleChangeStart,
    handleChangeFinish,
  } = props;

  const classes = styles();
  const theme = useTheme();
  const matchesSM = useMediaQuery(theme.breakpoints.down("sm"));
  const [openDrawer, setOpenDrawer] = useState(false);
  const iOS = process.browser && /iPad|iPhone|iPod/.test(navigator.userAgent);

  const [rowEnd, colEnd] = valToIndx(end, nCols);
  const [rowStart, colStart] = valToIndx(start, nCols);
  const [alpha, setAlpha] = useState(0.57);
  const [fName, setFName] = useState("A*");
  const [openAlert, setOpenAlert] = useState(false);

  const handleClickOpenAlert = () => {
    setOpenAlert(true);
  };

  const handleCloseAlert = () => {
    setOpenAlert(false);
  };

  const changeAlpha = (alpha) => setAlpha(alpha);

  const clear = () => {
    clearAnimation(grid.grid, start, end);
    const n = new Grid(start, end, nRows, nCols, wRange, MIN);
    setGrid(grid);
    setIsWeighted(false);
  };

  const newMaze = async () => {
    const newGrid = await getNewMazedGrid(grid, 0.33);
    setGrid(newGrid);
    // console.log(grid === newGrid)
  };

  const negativeWeight = () => {
    clearAnimation(grid, start, end);
    let n;
    if (!isNegative) {
      n = new Grid(start, end, nRows, nCols, wRange, MIN);
    } else {
      n = new Grid(start, end, nRows, nCols, wRange);
    }
    setGrid(n);
    setIsWeighted(true);
    toggleIsNegative();
  };

  const handleClick = async (alg) => {
    setDisable(true);
    await alg;
    setDisable(false);
  };

  const hanldeSliderChange = () => {
    if (alpha !== 0 && alpha !== 1) setFName("A*");
    else {
      if (alpha === 0) setFName("Best Fisrt Search");
      if (alpha === 1) setFName("Dijkstra");
    }
  };

  const btnOptList = [
    { name: "Clear", click: () => clear(), disabled: disable },
    {
      name: "Clear Path",
      click: () => clearPathAnimation(grid),
      disabled: disable,
    },
    { name: "Maze", click: () => handleClick(newMaze()), disabled: disable },
    {
      name: "Change Start",
      click: () => handleChangeStart(),
      disabled: disable,
    },
    {
      name: "Change Target",
      click: () => handleChangeFinish(),
      disabled: disable,
    },
    {
      name: isWeighted ? "Unweighted Grid" : "Weighted Grid",
      click: () => handleClick(toggleIsweighted()),
      disabled: disable,
    },
    {
      name: isNegative ? "Positive" : "Negative",
      click: negativeWeight,
      disabled: disable,
    },
  ];

  const unWBtnsList = [
    {
      name: "BFS",
      click: () => {
        console.log("bfs cliked");
        handleClick(
          bfs(
            grid,
            grid.grid[rowStart][colStart],
            grid.grid[rowEnd][colEnd],
            nCols
          )
        );
      },
      disabled: isWeighted ? true : disable,
    },
    {
      name: "DFS",
      click: () =>
        handleClick(
          dfs(grid, grid[rowStart][colStart], grid[rowEnd][colEnd], nCols)
        ),
      disabled: isWeighted ? true : disable,
    },
  ];

  const btnOpts = (
    <Fragment>
      {btnOptList.map((btn) =>
        btn.name === "Change Start" || btn.name === "Change Target" ? null : (
          <Button
            key={`btnOpt-${btn.name}`}
            className={classes.button}
            onClick={btn.click}
            disabled={btn.disabled}
          >
            {btn.name}
          </Button>
        )
      )}
    </Fragment>
  );

  const unWBtns = (
    <Fragment>
      {unWBtnsList.map((btn) => (
        <Button
          key={`unWBtn-${btn.name}`}
          className={classes.button}
          onClick={btn.click}
          disabled={btn.disabled}
        >
          <Typography className={classes.text}>{btn.name}</Typography>
        </Button>
      ))}
    </Fragment>
  );

  const wBtns = (
    <Fragment>
      <div className={classes.slider}>
        <Slider
          defaultValue={alpha}
          min={0}
          max={1}
          onAfterChange={hanldeSliderChange}
          onChange={changeAlpha}
          step={0.01}
          disabled={disable}
        />
      </div>
      <span className={classes.text}>Alpha: {alpha}</span>
      <Button
        className={classes.button}
        onClick={() =>
          handleClick(
            a(
              grid,
              grid[rowStart][colStart],
              grid[rowEnd][colEnd],
              nCols,
              wRange,
              alpha
            )
          )
        }
        disabled={disable}
      >
        {fName}
      </Button>
    </Fragment>
  );

  const negBtns = (
    <Fragment>
      <Button
        onClick={handleClickOpenAlert}
        disabled={disable}
        className={classes.button}
      >
        <Typography className={classes.text}>Bellman-Ford</Typography>
      </Button>
      <Dialog
        open={openAlert}
        onClose={handleCloseAlert}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">
          How to run Bellman-Ford?
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            * Recommended Mode: Will Clear Grid, add to layers of walls and then
            run Bellman-Ford. PS: Put both start and finsih node in the same
            connected component is necessary for search min Path.
          </DialogContentText>
          <DialogContentText>
            * Free-Style: Won't change any grid property and you can run as you
            want.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={async () => {
              handleCloseAlert();
              clear();
              await newMaze();
              await newMaze();
              setIsWeighted(true);
              handleClick(
                bellmanFord(
                  grid,
                  grid[rowStart][colStart],
                  grid[rowEnd][colEnd],
                  nCols
                )
              );
            }}
            color="primary"
          >
            Recommended
          </Button>
          <Button
            onClick={async () => {
              handleCloseAlert();
              handleClick(
                bellmanFord(
                  grid,
                  grid[rowStart][colStart],
                  grid[rowEnd][colEnd],
                  nCols
                )
              );
            }}
            color="primary"
            autoFocus
          >
            Free-Style
          </Button>
        </DialogActions>
      </Dialog>
    </Fragment>
  );

  const btns = (
    <Fragment>
      <div className={classes.btnOpt}> {btnOpts}</div>
      <div>{isWeighted ? null : unWBtns}</div>
      <div>{!isWeighted ? null : isNegative ? negBtns : wBtns}</div>
      <Button
        className={classes.button}
        onClick={() => {
          window.open(
            "https://github.com/LyangHiga/pathfinding-visualizer#instructions",
            "_blank"
          );
        }}
      >
        Instructions
      </Button>
    </Fragment>
  );

  const drawer = (
    <Fragment>
      <SwipeableDrawer
        disableBackdropTransition={!iOS}
        disableDiscovery={iOS}
        open={openDrawer}
        onClose={() => setOpenDrawer(false)}
        onOpen={() => setOpenDrawer(true)}
      >
        <List disablePadding>
          {btnOptList.map((btn) => (
            <ListItem
              key={`list-${btn.name}`}
              onClick={() => {
                setOpenDrawer(false);
                btn.click();
              }}
              divider
              button
              className={classes.button}
              disabled={disable}
            >
              <ListItemText>{btn.name}</ListItemText>
            </ListItem>
          ))}
        </List>
        {isWeighted ? (
          !isNegative ? (
            <List disablePadding>
              <ListItem divider>
                <Slider
                  defaultValue={alpha}
                  min={0}
                  max={1}
                  onAfterChange={hanldeSliderChange}
                  onChange={changeAlpha}
                  step={0.01}
                  disabled={disable}
                  className={classes.slider}
                />
              </ListItem>
              <ListItem divider className={classes.button}>
                <ListItemText>Alpha: {alpha}</ListItemText>
              </ListItem>
              <ListItem
                onClick={() => {
                  setOpenDrawer(false);
                  handleClick(
                    a(
                      grid,
                      grid[rowStart][colStart],
                      grid[rowEnd][colEnd],
                      nCols,
                      wRange,
                      alpha
                    )
                  );
                }}
                divider
                button
                className={classes.button}
                disabled={disable}
              >
                <ListItemText>{fName}</ListItemText>
              </ListItem>
            </List>
          ) : (
            <List disablePadding>
              <ListItem
                onClick={handleClickOpenAlert}
                divider
                button
                className={classes.button}
                disabled={disable}
              >
                <ListItemText>Bellman-Ford</ListItemText>
              </ListItem>
              <Dialog
                open={openAlert}
                onClose={handleCloseAlert}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
              >
                <DialogTitle id="alert-dialog-title">
                  How to run Bellman-Ford?
                </DialogTitle>
                <DialogContent>
                  <DialogContentText id="alert-dialog-description">
                    * Recommended Mode: Will Clear Grid, add to layers of walls
                    and then run Bellman-Ford. PS: Put both start and finsih
                    node in the same connected component is necessary for search
                    min Path.
                  </DialogContentText>
                  <DialogContentText>
                    * Free-Style: Won't change any grid property and you can run
                    as you want.
                  </DialogContentText>
                </DialogContent>
                <DialogActions>
                  <Button
                    onClick={async () => {
                      setOpenDrawer(false);
                      handleCloseAlert();
                      clear();
                      await newMaze();
                      await newMaze();
                      setIsWeighted(true);
                      handleClick(
                        bellmanFord(
                          grid,
                          grid[rowStart][colStart],
                          grid[rowEnd][colEnd],
                          nCols
                        )
                      );
                    }}
                    color="primary"
                  >
                    Recommended
                  </Button>
                  <Button
                    onClick={async () => {
                      setOpenDrawer(false);
                      handleCloseAlert();
                      handleClick(
                        bellmanFord(
                          grid,
                          grid[rowStart][colStart],
                          grid[rowEnd][colEnd],
                          nCols
                        )
                      );
                    }}
                    color="primary"
                    autoFocus
                  >
                    Free-Style
                  </Button>
                </DialogActions>
              </Dialog>
            </List>
          )
        ) : (
          <List disablePadding>
            {unWBtnsList.map((btn) => (
              <ListItem
                key={`list-${btn.name}`}
                onClick={() => {
                  setOpenDrawer(false);
                  btn.click();
                }}
                divider
                button
                className={classes.button}
                disabled={disable}
              >
                <ListItemText>{btn.name}</ListItemText>
              </ListItem>
            ))}
          </List>
        )}
        <List>
          <ListItem
            divider
            className={classes.button}
            onClick={() => {
              window.open(
                "https://github.com/LyangHiga/pathfinding-visualizer#instructions",
                "_blank"
              );
            }}
          >
            <ListItemText>Instructions</ListItemText>
          </ListItem>
        </List>
      </SwipeableDrawer>
      <IconButton
        className={classes.drawerIconContainer}
        onClick={() => setOpenDrawer(!openDrawer)}
        disableRipple
      >
        <MenuIcon></MenuIcon>
      </IconButton>
    </Fragment>
  );

  return (
    <AppBar position="static" color="inherit" className={classes.Navbar}>
      <Toolbar>
        <Typography className={classes.title} variant="h6" color="inherit">
          Pathfinding Visualizer
        </Typography>
        {matchesSM ? drawer : btns}
      </Toolbar>
    </AppBar>
  );
}

export default Nav;
