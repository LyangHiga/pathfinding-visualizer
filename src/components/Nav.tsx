// TODO: Refactoring Drawer and btns, break down in smaller components
// TODO: Refactoring with context api, too much props
import React, { Fragment, useState } from "react";
import "rc-slider/assets/index.css";
import bfs from "../algorithms/bfs";
import dfs from "../algorithms/dfs";

import { clearAnimation, clearPathAnimation } from "../helpers/animations";

import AppBar from "@material-ui/core/AppBar";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { useTheme } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import SwipeableDrawer from "@material-ui/core/SwipeableDrawer";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";

import styles from "../styles/NavStyles";
import { valToIndx, getNewMazedGrid } from "../helpers/gridHelper";

import Grid from "../models/Grid";
import { NavProps, FunctionHandled } from "./types/NavTypes";
import WeightedButton from "./WeightedButton";
import NegButton from "./NegButton";

function Nav(props: NavProps) {
  const {
    grid,
    disable,
    start,
    target,
    setGrid,
    setDisable,
    isWeighted,
    setIsWeighted,
    toggleIsweighted,
    isNegative,
    toggleIsNegative,
    handleChangeStart,
    handleChangeTarget,
  } = props;

  const { nCols, nRows, max, min } = grid;

  const classes = styles();
  const theme = useTheme();
  const matchesSM = useMediaQuery(theme.breakpoints.down("sm"));
  const [openDrawer, setOpenDrawer] = useState(false);

  const [rowTarget, colTarget] = valToIndx(target, nCols);
  const [rowStart, colStart] = valToIndx(start, nCols);

  const clear = () => {
    clearAnimation(grid, start, target);
    const n = new Grid(start, target, nRows, nCols, max, min);
    setGrid(n);
    setIsWeighted(false);
  };

  const newMaze = async () => {
    const newGrid = await getNewMazedGrid(grid, 0.33);
    setGrid(newGrid);
    // console.log(grid === newGrid)
  };

  const negativeWeight = () => {
    clearAnimation(grid, start, target);
    let n;
    if (!isNegative) {
      n = new Grid(start, target, nRows, nCols, max, min);
    } else {
      n = new Grid(start, target, nRows, nCols, max);
    }
    setGrid(n);
    setIsWeighted(true);
    toggleIsNegative();
  };

  const handleClick = async (alg: FunctionHandled) => {
    setDisable(true);
    await alg;
    setDisable(false);
  };

  const btnOptList = [
    { name: "Clear", click: () => clear(), disabled: disable },
    {
      name: "Clear Path",
      click: () => clearPathAnimation(grid),
      disabled: disable,
    },
    { name: "Maze", click: () => handleClick(newMaze()), disabled: disable },
    {
      name: "Change Start",
      click: () => handleChangeStart(),
      disabled: disable,
    },
    {
      name: "Change Target",
      click: () => handleChangeTarget(),
      disabled: disable,
    },
    {
      name: isWeighted ? "Unweighted Grid" : "Weighted Grid",
      click: () => toggleIsweighted(),
      disabled: disable,
    },
    {
      name: isNegative ? "Positive" : "Negative",
      click: negativeWeight,
      disabled: disable,
    },
  ];

  const unWBtnsList = [
    {
      name: "BFS",
      click: () => {
        handleClick(
          bfs(
            grid,
            grid.grid[rowStart][colStart],
            grid.grid[rowTarget][colTarget]
          )
        );
      },
      disabled: isWeighted ? true : disable,
    },
    {
      name: "DFS",
      click: () =>
        handleClick(
          dfs(
            grid,
            grid.grid[rowStart][colStart],
            grid.grid[rowTarget][colTarget]
          )
        ),
      disabled: isWeighted ? true : disable,
    },
  ];

  const btnOpts = (
    <Fragment>
      {btnOptList.map((btn) =>
        btn.name === "Change Start" || btn.name === "Change Target" ? null : (
          <Button
            key={`btnOpt-${btn.name}`}
            className={classes.button}
            onClick={btn.click}
            disabled={btn.disabled}
          >
            {btn.name}
          </Button>
        )
      )}
    </Fragment>
  );

  const unWBtns = (
    <Fragment>
      {unWBtnsList.map((btn) => (
        <Button
          key={`unWBtn-${btn.name}`}
          className={classes.button}
          onClick={btn.click}
          disabled={btn.disabled}
        >
          <Typography className={classes.text}>{btn.name}</Typography>
        </Button>
      ))}
    </Fragment>
  );

  const btns = (
    <Fragment>
      <div className={classes.btnOpt}> {btnOpts}</div>
      <div>{isWeighted ? null : unWBtns}</div>
      <div>
        {!isWeighted ? null : isNegative ? (
          <NegButton
            grid={grid}
            start={start}
            target={target}
            disable={disable}
            btn={true}
            handleClick={handleClick}
            setOpenDrawer={setOpenDrawer}
            clear={clear}
            newMaze={newMaze}
            setIsWeighted={setIsWeighted}
          />
        ) : (
          <WeightedButton
            grid={grid}
            start={start}
            target={target}
            disable={disable}
            btn={true}
            handleClick={handleClick}
            setOpenDrawer={setOpenDrawer}
          />
        )}
      </div>
      <Button
        className={classes.button}
        onClick={() => {
          window.open(
            "https://github.com/LyangHiga/pathfinding-visualizer#instructions",
            "_blank"
          );
        }}
      >
        Instructions
      </Button>
    </Fragment>
  );

  const drawer = (
    <Fragment>
      <SwipeableDrawer
        open={openDrawer}
        onClose={() => setOpenDrawer(false)}
        onOpen={() => setOpenDrawer(true)}
      >
        <List disablePadding>
          {btnOptList.map((btn) => (
            <ListItem
              key={`list-${btn.name}`}
              onClick={() => {
                setOpenDrawer(false);
                btn.click();
              }}
              divider
              button
              className={classes.button}
              disabled={disable}
            >
              <ListItemText>{btn.name}</ListItemText>
            </ListItem>
          ))}
        </List>
        {isWeighted ? (
          !isNegative ? (
            <WeightedButton
              grid={grid}
              start={start}
              target={target}
              disable={disable}
              btn={false}
              handleClick={handleClick}
              setOpenDrawer={setOpenDrawer}
            />
          ) : (
            <NegButton
              grid={grid}
              start={start}
              target={target}
              disable={disable}
              btn={false}
              handleClick={handleClick}
              setOpenDrawer={setOpenDrawer}
              clear={clear}
              newMaze={newMaze}
              setIsWeighted={setIsWeighted}
            />
          )
        ) : (
          <List disablePadding>
            {unWBtnsList.map((btn) => (
              <ListItem
                key={`list-${btn.name}`}
                onClick={() => {
                  setOpenDrawer(false);
                  btn.click();
                }}
                divider
                button
                className={classes.button}
                disabled={disable}
              >
                <ListItemText>{btn.name}</ListItemText>
              </ListItem>
            ))}
          </List>
        )}
        <List>
          <ListItem
            divider
            className={classes.button}
            onClick={() => {
              window.open(
                "https://github.com/LyangHiga/pathfinding-visualizer#instructions",
                "_blank"
              );
            }}
          >
            <ListItemText>Instructions</ListItemText>
          </ListItem>
        </List>
      </SwipeableDrawer>
      <IconButton
        className={classes.drawerIconContainer}
        onClick={() => setOpenDrawer(!openDrawer)}
        disableRipple
      >
        <MenuIcon></MenuIcon>
      </IconButton>
    </Fragment>
  );

  return (
    <AppBar position="static" color="inherit" className={classes.Navbar}>
      <Toolbar>
        <Typography className={classes.title} variant="h6" color="inherit">
          Pathfinding Visualizer
        </Typography>
        {matchesSM ? drawer : btns}
      </Toolbar>
    </AppBar>
  );
}

export default Nav;
