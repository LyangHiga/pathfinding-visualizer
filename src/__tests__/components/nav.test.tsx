import React from "react";
import { shallow } from "enzyme";
import Nav from "../../components/Nav";
import Grid from "../../models/Grid";

const START = 10;
const TARGET = 14;

const NCOLS = 5;
const NROWS = 5;

//       0  1   2   3   4
//       5  6   7   8   9
// g =   S  11  12  13  T
//      15  16  17  18  19
//      20  21  22  23  24

describe("Node", () => {
  it("renders correctly", () => {
    const grid = new Grid(START, TARGET, NROWS, NCOLS);
    const wrapper = shallow(
      <Nav
        grid={grid}
        disable={false}
        start={START}
        target={TARGET}
        isWeighted={false}
        isNegative={false}
        setGrid={jest.fn()}
        setDisable={jest.fn()}
        setIsWeighted={jest.fn()}
        toggleIsweighted={jest.fn()}
        toggleIsNegative={jest.fn()}
        handleChangeStart={jest.fn()}
        handleChangeTarget={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will generate a snapshot file automatically.
  });
});
