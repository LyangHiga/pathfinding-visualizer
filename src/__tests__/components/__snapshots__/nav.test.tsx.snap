// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-Navbar-1"
  color="inherit"
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      color="inherit"
      variant="h6"
    >
      Pathfinding Visualizer
    </WithStyles(ForwardRef(Typography))>
    <div
      className="makeStyles-btnOpt-3"
    >
       
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="btnOpt-Clear"
        onClick={[Function]}
      >
        Clear
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="btnOpt-Clear Path"
        onClick={[Function]}
      >
        Clear Path
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="btnOpt-Maze"
        onClick={[Function]}
      >
        Maze
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="btnOpt-Weighted Grid"
        onClick={[Function]}
      >
        Weighted Grid
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="btnOpt-Negative"
        onClick={[Function]}
      >
        Negative
      </WithStyles(ForwardRef(Button))>
    </div>
    <div>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="unWBtn-BFS"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-text-7"
        >
          BFS
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-4"
        disabled={false}
        key="unWBtn-DFS"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-text-7"
        >
          DFS
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Button))>
    </div>
    <div />
    <WithStyles(ForwardRef(Button))
      className="makeStyles-button-4"
      onClick={[Function]}
    >
      Instructions
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
